src/Collection.php

```diff
@@ -520,9 +520,10 @@ public function find($filter = [], array $options = [])
             $options['typeMap'] = $this->typeMap;
         }
 
-        $operation = new Find($this->databaseName, $this->collectionName, $filter, $options);
+        //$operation = new Find($this->databaseName, $this->collectionName, $filter, $options);
+        //return $operation->execute($server);
 
-        return $operation->execute($server);
+        return new \MongoDB\Cursor($this, $filter, $options);
```

Comment:
That's true. I was unsure to do it in constructor or not, but now i'm gonna execute it in the constructor method. Thanks for the review.